server/visualizer/handler.go

```diff
@@ -0,0 +1,260 @@
+			return d.Field(h.DataModelDataFieldID)
+		})
+		return c.JSON(http.StatusOK, data2)
+	}
+}
+
+// GET | /viz/:pid/data/:iid
+func (h *Handler) getDataHandler() func(c echo.Context) error {
+	return func(c echo.Context) error {
+		ctx := c.Request().Context()
+		itemID := c.Param("iid")
+		if itemID == "" {
+			return c.JSON(http.StatusNotFound, nil)
+		}
+
+		data, err := h.CMS.GetItem(ctx, itemID)
+		if err != nil {
+			return err
+		}
+		return c.JSON(http.StatusOK, data.Field(h.DataModelIDFieldID).Value)
+	}
+}
+
+// POST | /viz/:pid/data
+func (h *Handler) createDataHandler() func(c echo.Context) error {
+	return func(c echo.Context) error {
+		ctx := c.Request().Context()
+		b, err := io.ReadAll(c.Request().Body)
+		if err != nil {
+			return fmt.Errorf("occur an unexpected EOF error: %w", err)
+		}
+
+		fields := []cms.Field{{
+			ID:    h.DataModelDataFieldID,
+			Value: string(b),
+		}}
+		item, err := h.CMS.CreateItem(ctx, h.DataModelKey, fields)
+		if err != nil {
+			return err
+		}
+
+		return c.JSON(http.StatusOK, item.Field(h.DataModelDataFieldID).Value)
+	}
+}
+
+// PATCH | /viz/:pid/data/:did
+func (h *Handler) updateDataHandler() func(c echo.Context) error {
+	return func(c echo.Context) error {
+		ctx := c.Request().Context()
+		itemID := c.Param("iid")
+		b, err := io.ReadAll(c.Request().Body)
+		if err != nil {
+			return fmt.Errorf("occur an unexpected EOF error: %w", err)
+		}
+
+		fields := []cms.Field{{
+			ID:    h.DataModelDataFieldID,
+			Value: string(b),
+		}}
+
+		item, err := h.CMS.UpdateItem(ctx, itemID, fields)
+		if err != nil {
+			return err
+		}
+
+		return c.JSON(http.StatusOK, item.Field(h.DataModelDataFieldID).Value)
+	}
+}
+
+// DELETE | /viz/:pid/data/:did
+func (h *Handler) deleteDataHandler() func(c echo.Context) error {
+	return func(c echo.Context) error {
+		ctx := c.Request().Context()
+		itemID := c.Param("iid")
+
+		err := h.CMS.DeleteItem(ctx, itemID)
+		if err != nil {
+			return err
+		}
+
+		return c.JSON(http.StatusOK, nil)
+	}
+}
+
+// GET | /viz/:id/templates
+func (h *Handler) fetchTemplate() func(c echo.Context) error {
+	return func(c echo.Context) error {
+		ctx := c.Request().Context()
+		templates, err := h.CMS.GetItems(ctx, h.TemplateModelKey)
+		if err != nil {
+			return err
+		}
+
+		templates2 := lo.Map(templates.Items, func(t cms.Item, _ int) any {
+			return t.Field(h.TemplateModelTemplateFieldID).Value
+		})
+		return c.JSON(http.StatusOK, templates2)
+	}
+}
+
+// POST | /viz/:pid/templates
+func (h *Handler) createTemplateHandler() func(c echo.Context) error {
+	return func(c echo.Context) error {
+		ctx := c.Request().Context()
+		b, err := io.ReadAll(c.Request().Body)
+		if err != nil {
+			return fmt.Errorf("occur an unexpected EOF error: %w", err)
+		}
+
+		fields := []cms.Field{{
+			ID:    h.TemplateModelTemplateFieldID,
+			Value: string(b),
+		}}
+		item, err := h.CMS.CreateItem(ctx, h.TemplateModelKey, fields)
```

Comment:
```suggestion
		item, err := h.CMS.CreateItemByKey(ctx, h.CMSProject, h.TemplateModelKey, fields)
```
Also an proejct ID is requried to create an item by a model key.

the same applies hereafter